slime/backends/megatron_utils/model.py

```diff
@@ -110,10 +109,99 @@ def setup_model_and_optimizer(
         skip_load_to_model_and_opt=False,
     )
 
+    warmup(args, model, optimizer)
+
     print(f"Loaded checkpoint at iteration {iteration}")
     return model, optimizer, opt_param_scheduler, iteration
 
 
+def warmup(args, model, optimizer):
```

Comment:
curious whether we can put all training steps under mem_saver.region()

Reply:
yeah, I believe this is a better idea than `warmup`. I will remove the warmup here and try to put the whole training procedure under `mem_saver.region()`.

Reply:
sure! if you find any errors just ping me (e.g. I expect some tensors may be used for cross-gpu communication and you need torch_memory_saver.disable() on it).